go/arrow/csv/reader.go

```diff
@@ -755,6 +759,18 @@ func (r *Reader) parseList(field array.Builder, str string) {
 	}
 }
 
+func (r *Reader) parseBinaryDataType(field array.Builder, str string) {
+	if r.isNull(str) {
+		field.AppendNull()
+		return
+	}
+	decodedVal, err := base64.StdEncoding.DecodeString(str)
```

Comment:
do we want to check both the padded and non-padded encodings? (ie: both `StdEncoding` and `RawStdEncoding`, checking the other if the first one errors)

If we don't check both encodings, we should make sure the documentation states which we use.

Reply:
I would stay with one option for starters. Where should I document it?

Reply:
Hmm. I'd suggest adding it to the `NewReader` godoc comment as that seems most reasonable to me. To mention that a binary field expects to be base64 encoded with padding as per `base64.StdEncoding`